app/src/test/java/nl/tudelft/b_b_w/blockchain/BlockTest.java

```diff
@@ -0,0 +1,149 @@
+package nl.tudelft.b_b_w.blockchain;
+
+import net.i2p.crypto.eddsa.EdDSAPrivateKey;
+import net.i2p.crypto.eddsa.EdDSAPublicKey;
+import net.i2p.crypto.eddsa.Utils;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import nl.tudelft.b_b_w.controller.ED25519;
+import nl.tudelft.b_b_w.model.TrustValues;
+
+import static junit.framework.Assert.assertEquals;
+import static junit.framework.Assert.assertFalse;
+import static junit.framework.Assert.assertTrue;
+
+/**
+ * Junit Testing class of the Block class
+ */
+public class BlockTest {
+    /**
+     * Properties of the owner of the block
+     */
+    private User owner;
+    private String ownerName;
+    private String ownerIban;
+    private EdDSAPublicKey ownerPublicKey;
+
+
+    /**
+     * Properties of the contact/ block sender
+     */
+    private User contact;
+    private String contactName;
+    private String contactIban;
+    private EdDSAPublicKey contactPublicKey;
+
+    /**
+     * Properties of a BlockData object
+     */
+    private BlockData blockData;
+    private BlockType blockType;
+    private int sequenceNumber;
+    private Hash previousHashChain;
+    private Hash previousHashSender;
+    private double trustValue;
+
+    private Hash testOwnHash;
+
+    /**
+     * The block.
+     */
+    private Block block;
+    private Block genesisBlock;
+
+    /**
+     * Set up the before testing.
+     */
+    @Before
+    public void setUpBlock() throws Exception {
+
+
+        //setting up owner
+        ownerName = "BlockOwner1";
+        ownerIban = "Owner1Iban";
+        //object to generate public key
+        EdDSAPrivateKey edDSAPrivateKey1 =
+                ED25519.generatePrivateKey(Utils.hexToBytes("0000000000000000000000000000000000000000000000000000000000000000"));
```

Comment:
I have edited these in other test files. But for this particular test of equals, the own hash is being compared with a predefined value, so if the private key is not the same every time the test run the hash will be different and the test will fail.